apps/rush-lib/src/schemas/command-line.schema.json

```diff
@@ -149,19 +149,117 @@
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            }
+          }
+        },
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "commandKind": { "$ref": "#/definitions/anything" },
+            "name": { "$ref": "#/definitions/anything" },
+            "summary": { "$ref": "#/definitions/anything" },
+            "description": { "$ref": "#/definitions/anything" },
+            "safeForSimultaneousRushProcesses": { "$ref": "#/definitions/anything" },
+
+            "phases": { "$ref": "#/definitions/anything" }
+          }
+        }
+      ]
+    },
+
+    "phase": {
+      "title": "Phase",
+      "description": "A phase, used in the phased command feature.",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["name"],
+      "properties": {
+        "name": {
+          "title": "Name",
+          "description": "The name of the phase. Note that this value must start with the \"_phase:\" prefix.",
+          "type": "string"
+        },
+
+        "dependencies": {
+          "title": "Dependencies",
+          "description": "The dependencies of this phase.",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "self": {
+              "title": "Self",
+              "description": "Dependency phases within the same project.",
+              "type": "array",
+              "uniqueItems": true,
+              "items": {
+                "type": "string"
+              }
+            },
+            "upstream": {
+              "title": "Upstream",
+              "description": "Dependency phases in upstream projects.",
+              "type": "array",
+              "uniqueItems": true,
+              "items": {
+                "type": "string"
+              }
+            }
+          }
+        },
+
+        "enableParallelism": {
+          "title": "enableParallelism",
+          "description": "If true then this phase can be run in parallel, i.e. executed simultaneously for multiple projects.",
+          "type": "boolean"
+        },
+
+        "ignoreMissingScript": {
+          "title": "Ignore Missing Script",
+          "description": "Normally Rush requires that each project's package.json has a \"scripts\" entry matching the phase name. To disable this check, set \"ignoreMissingScript\" to true.",
+          "type": "boolean"
+        },
+
+        "allowWarningsOnSuccess": {
+          "title": "Allow Warnings on Success",
+          "description": "By default, Rush returns a nonzero exit code if errors or warnings occur during a command. If this option is set to \"true\", Rush will return a zero exit code if warnings occur during the execution of this phase.",
+          "type": "boolean"
+        }
+      }
+    },
```

Comment:
For phased commands, just looking at the list of properties moved over from the command to the phase:

### incremental

Is `incremental` property missing for phased commands? I don't see it on either the command or the phase. (We discussed assuming that the build cache would be enabled first, but I'm not sure if that means we don't want the ability to _force_ a command not to be incremental -- otherwise how can we support rebuild.)

### enableParallelism

On a phase, `enableParallelism` means that two instances of _that phase_ have to run sequentially (while other phases can run at the same time), or that the phase must be run "alone" (blocking ALL other phases)?

I guess I can see it working either way, and I almost wonder if we need to consider both options.

### ignoreMissingScript

Makes sense. In my initial discussion with @octogonz I was very worried about "onboarding" -- how to slowly bring up projects in a monorepo with lots of project owners and different build frameworks. But, as long as you're allowed to define a phase as "" (empty string), the easy way to onboard is just collapse everything to the first phase.

For example, a new phased system with 3 phases (`compile`, `lint`, `test`), you could copy/paste this boilerplate into every single project's `package.json`:

```js
    "_phase:compile": "[insert entire contents of 'build' script]",
    "_phase:lint": "",
    "_phase:test": ""
```

This way the entire monorepo is immediately valid, even if you haven't figured out how to break apart "compile" and "lint" for every project yet (or even if there are some projects that CANNOT break apart compile and lint).

### allowWarningsOnSuccess

:+1: A net win for phased builds, especially if you're one of those "don't care about lint warnings" people.  Now you can ignore warnings just for one phase of a build instead of the entire build.